generic/sosreport.py

```diff
@@ -0,0 +1,149 @@
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Pavithra <pavrampu@linux.vnet.ibm.com>
+
+import os
+import tempfile
+import shutil
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class sosreport_test(Test):
+
+    is_fail = 0
+
+    def run_cmd(self, cmd):
+        self.log.info("executing ============== %s =================" % cmd)
+        if process.system(cmd, ignore_status=True, sudo=True):
+            self.log.info("%s command failed" % cmd)
+            self.is_fail += 1
+        return
+
+    def run_cmd_out(self, cmd):
+        return process.system_output(cmd, shell=True, ignore_status=True, sudo=True)
+
+    def setUp(self):
+        dist = distro.detect()
+        sm = SoftwareManager()
+        if not process.system("sosreport", ignore_status=True, sudo=True):
+            self.log.info("sosreport is installed")
+        elif dist.name == 'Ubuntu':
+            if not sm.install("sosreport"):
+                self.skip("Fail to install sosreport required for this test.")
+        elif dist.name == 'redhat':
+            if not sm.install("sos"):
+                self.skip("Fail to install sos package required for this test.")
+        else:
+            self.skip("sosreport is not supported on this distro "
+                      "or package is not installed")
+
+    def test(self):
+        self.log.info(
+            "===============Executing sosreport tool test===============")
+        directory_name = tempfile.mkdtemp()
+        self.run_cmd("sosreport -h")
+        self.run_cmd("sosreport -l")
+        list = self.params.get('list', default=['--all-logs'])
+        for list_item in list:
+            cmd = "sosreport --batch --tmp-dir=%s %s" % (directory_name, list_item)
+            self.run_cmd(cmd)
+        if self.run_cmd_out("sosreport --batch --tmp-dir=%s -n libraries | "
+                            "grep libraries" % directory_name):
+            self.is_fail += 1
+            self.log.info("--skip-plugins option failed")
+        self.run_cmd("sosreport --batch --tmp-dir=%s -e ntp,numa,snmp" % directory_name)
+        if 'sendmail' not in self.run_cmd_out("sosreport --batch --tmp-dir=%s -e "
+                                              "sendmail | grep sendmail" % directory_name):
+            self.is_fail += 1
+            self.log.info("--enable-plugins option failed")
+        self.run_cmd("sosreport --batch --tmp-dir=%s -o "
+                     "pci,powerpc,procenv,process,processor,kdump" % directory_name)
```

Comment:
this command only run in power pc if yes please have check of pate form before run this test as in other arch it kind of misleading